### PR TITLE
Skip ECC bounds usage when converting probabilities to percentiles

### PR DESCRIPTION
**Pull request by gavinevans**
Related to: https://github.com/metoppv/mo-blue-team/issues/507, https://github.com/metoppv/improver/pull/1838
Acceptance test data: https://github.com/metoppv/improver_test_data/pull/23

Description
This PR adds support, similar to #1838, to suppress the usage of ECC bounds when converting probabilities to percentiles. This avoids the percentiles at either end of the distribution being dominating by the ECC bound and hence having highly unrealistic values. 

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py [hunk 1]
```diff
@@ -506,14 +471,14 @@ def test_masked_data_below(self):
         cube = set_up_probability_cube(
             1 - self.cube.data,
             [200, 1000, 15000],
-            variable_name="cloud_base_height_assuming_only_consider_cloud_\
-                area_fraction_greater_than_4p5_oktas",
+            variable_name="cloud_base_height_assuming_only_consider_cloud_"
+            + "area_fraction_greater_than_4p5_oktas",
```
Comment (bayliffe) [inline improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py:475]: Seems a bit retro.
```suggestion
            variable_name=(
                "cloud_base_height_assuming_only_consider_cloud_"
                "area_fraction_greater_than_4p5_oktas"
            ),
```

File: improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py [hunk 2]
```diff
@@ -531,14 +496,14 @@ def test_masked_data_above(self):
         cube = set_up_probability_cube(
             self.cube.data,
             [200, 1000, 15000],
-            variable_name="cloud_base_height_assuming_only_consider_cloud_\
-                area_fraction_greater_than_4p5_oktas",
+            variable_name="cloud_base_height_assuming_only_consider_cloud_"
+            + "area_fraction_greater_than_4p5_oktas",
```
Comment (bayliffe) [inline improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py:500]:
```suggestion
            variable_name=(
                "cloud_base_height_assuming_only_consider_cloud_"
                "area_fraction_greater_than_4p5_oktas"
            ),
```

**Review by bayliffe — DISMISSED** [review]
Looks fine Gavin. Some completely innocuous comments, but nothing of note.

**Review by Katie-Howard — APPROVED** [review]
I've looked through the code changes and I think it's all doing what it's supposed to.  I've run the acceptance and unit tests and they all seem to pass too, so I'm happy with it all.

File: improver_tests/acceptance/test_generate_percentiles.py [hunk 3]
```diff
@@ -79,7 +79,7 @@ def test_probconvert(tmp_path, count):
 
 
 @pytest.mark.slow
-def test_eccbounds(tmp_path):
+def test_ignoreeccbounds(tmp_path,):
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply improver_tests/acceptance/test_generate_percentiles.py]: Amended.

File: improver_tests/acceptance/test_generate_percentiles.py [hunk 4]
```diff
@@ -100,6 +100,35 @@ def test_eccbounds(tmp_path):
...
+        perc_input,
+        "--output",
+        output_path,
+        "--coordinates",
+        "realization",
+        "--percentiles",
+        "2,50,98",
+    ]
+    if bounds_option:
+        args += [bounds_option]
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply improver_tests/acceptance/test_generate_percentiles.py]: OK, I hadn't thought about that syntax. I've adopted that in the PR.

File: improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py (hunk 1 above)
Comment (gavinevans) [reply improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py]: Updated as suggested.

File: improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py (hunk 2 above)
Comment (gavinevans) [reply improver_tests/ensemble_copula_coupling/test_ConvertProbabilitiesToPercentiles.py]: Updated as suggested.

File: improver_tests/acceptance/SHA256SUMS [hunk 5]
```diff
@@ -337,6 +337,10 @@ a57f0db9012544e71e4ef322012bf5a4d03b4b84a2b03223446c66a9afe52a31  ./generate-per
 2b5a626ce6ccb3d2bfe393f7145997ff8d9bf92da8de8fcc461fbcab59cd79aa  ./generate-percentiles/probability_convert/multi_realization_kgo.nc
 6216e900501d35d7bc277791f0c5f343a33b44305121744ac7518ea87e6f6c4a  ./generate-percentiles/probability_convert/single_realization.nc
 40dd552c977743986a980da8eab7a03020049e2f29389467b7376e56037143dd  ./generate-percentiles/probability_convert/single_realization_kgo.nc
+f8aa2012e0796ed977262d269f22ab1fd96865a8d51094d575e38570c0a7cfd0  ./generate-percentiles/skip_ecc_bounds/input.nc
+55650cb8f8d3becc9f84a34c007ccda4f61b5908229584928c97d4d1cc997585  ./generate-percentiles/skip_ecc_bounds/input_old.nc
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply improver_tests/acceptance/SHA256SUMS]: I've removed this file from the checksums.

**Review by Katie-Howard — APPROVED** [review]
All the changes are fine (though you do need to delete skip_ecc_bounds/input_old.nc for the checksums to be right), so I'm happy with these changes 👍

**Review by bayliffe — APPROVED** [review]

**Pull request merged**